libass/ass_render.c

```diff
@@ -2302,12 +2302,20 @@ static void render_and_combine_glyphs(ASS_Renderer *render_priv,
 
     for (int i = 0; i < nb_bitmaps; i++) {
         CombinedBitmapInfo *info = &combined_info[i];
+        if (!info->bitmap_count)
+            continue;
+
+        int x_min = INT_MAX;
         for (int j = 0; j < info->bitmap_count; j++) {
+            if (info->bitmaps[j].bm)
+                x_min = FFMIN(x_min, info->bitmaps[j].pos.x + info->bitmaps[j].bm->left);
```

Comment:
I think karaoke should use outline coordinates regardless of whether we succeeded in creating bitmaps for them.

Reply:
I think so too, but it's quite hard to come up with good values that work sane with transformations. Also there is `first_pos_x` that is used as boolean flag. That version is at least looks better and fixes some issues.

Reply:
That should be some screen-space value so any outline parameter doesn't work. `bm->left` works better than before but the best solution will be, of course, to port that calculation from VSFilter